### ISSUE TITLE
Query settings not working with v2.5 and downgrade not solve the issue

### ISSUE THREAD

**Issue by momo-fr** [elementor/elementor#7662]
![bug-query](https://user-images.githubusercontent.com/29410831/55218828-81593f00-5203-11e9-845d-1e2d133113ef.jpg)

## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description
With Elementor v2.5 the widgets Products and Posts not work correctly. The query settings are no longer functional, category view is incomplete or empty, avoid duplicate no longer works.
I have dowgrade Elementor in 4.8.2 and I see that the function avoid duplicate no longer works but the category management is correct.

## Steps to reproduce
Add a widget Products, set the query on a specific category, duplicate the widget and change the query. Use different category or add another and set avoid duplicate on. With stabdard WP theme
I have some duplicate product and partial products display. If I downgrade on v2.8.4 the categories is complete but avoid duplicate not functionnal now.

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v5.6.42-84.2
	PHP Version: 7.2.16
	PHP Max Input Vars: 1000
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.1.1
	Site URL: https://meetup.freepixel.net
	Home URL: https://meetup.freepixel.net
	WP Multisite: No
	Max Upload Size: 128 MB
	Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: Europe/Paris
	Debug Mode: Inactive

== Theme ==
	Name: Twenty Nineteen
	Version: 1.3
	Author: the WordPress team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:66.0) Gecko/20100101 Firefox/66.0

== Active Plugins ==
	Elementor
		Version: 2.5.10
		Author: Elementor.com

	Elementor Pro
		Version: 2.5.2
		Author: Elementor.com

	WooCommerce
		Version: 3.5.7
		Author: Automattic

	Wordfence Security
		Version: 7.2.4
		Author: Wordfence


== Must-Use Plugins ==
	No french punctuation and accents for filename
		Version: 1.0
		Author: BeAPI


== Log ==
	: 
Log: showing 20 of 382019-03-24 14:41:53 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.0-beta2',
  'to' => '2.5.0-beta3',
)]
2019-03-27 08:38:12 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '2.5.9',
  'to' => '2.5.10',
)]
2019-03-27 08:38:13 [info] elementor-pro::elementor_pro_updater Started 
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_form Start  
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_form Finished 
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_woocommerce_menu_cart Start  
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_woocommerce_menu_cart Finished 
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_posts Start  
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_posts Finished 
2019-03-27 08:38:14 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.0-beta3',
  'to' => '2.5.1',
)]
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_portfolio Start  
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_portfolio Finished 
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_products Start  
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_products Finished 
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_sitemap Start  
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_sitemap Finished 
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_popup_border_radius Start  
2019-03-27 08:38:14 [info] Elementor Pro/Upgrades - _v_2_5_0_popup_border_radius Finished 
2019-03-27 08:38:14 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.0-beta3',
  'to' => '2.5.1',
)]
2019-03-28 07:45:23 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '2.5.1',
  'to' => '2.5.2',
)]

PHP: showing 3 of 3PHP: 2019-03-24 14:43:15 [notice X 1][/home/pixelwebftp/meetup/wp-content/plugins/elementor/includes/base/controls-stack.php::714] Undefined index: type [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-03-25 20:56:15 [notice X 14][/home/pixelwebftp/meetup/wp-content/plugins/elementor-pro/modules/posts/widgets/portfolio.php::632] Trying to get property 'term_id' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2019-03-28 09:10:34 [notice X 19][/home/pixelwebftp/meetup/wp-content/plugins/elementor-pro/modules/woocommerce/widgets/add-to-cart.php::130] Undefined index: post_id [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 1 of 1JS: 2019-03-28 14:10:19 [error X 7][https://meetup.freepixel.net/wp-content/plugins/elementor/assets/js/editor.min.js?ver=2.5.10:2:53410] this.ui.sliders.filter(...)[0] is undefined 
```
</details>

**Comment by momo-fr** [comment]
You can see here (downgrade 2.8.4): https://meetup.freepixel.net/propoqz/

**Comment by derweltbuerger** [comment]
Yes, indeed, I can confirm this bug in 2.5.1 and 2.5.2. I have several Product Archive Templates with query source set to 'latest products', 'include by' is set to 'term' (there's no other options available besides this one) and term itself is set to a specific product category that I want to display. With 2.4.8 the products are displayed but with 2.5.1 and 2.5.2 no products are displayed at all. This seems to be a major bug since I could not find a way to include products by category. I also tried deleting the widget and setting it up from scratch but to no avail.

Including single products by manual selection works, however.

Reverting to 2.4.8 helped for now. Please fix. Thanks!

**Comment by shilo-ey** [comment]
I merged this issue with #7653, which is on the same subject so we could deal with it more efficiently. 
Please feel free to continue the conversation there.

Thanks for your feedback!
